### PR TITLE
Add Python delta source and destination

### PR DESCRIPTION
**Pull request by JamesKnBr**

### CODE REVIEW

File: environment.yml [hunk 1]
```diff
@@ -70,4 +70,7 @@ dependencies:
         - deltalake==0.10.0
         - moto[s3]==4.1.13
         - build==0.10.0
+        - polars==0.18.8
+        - kafka-python==2.0.2
```
Comment (GBBBAS) [inline environment.yml:74]: Can we remove this?

File: src/sdk/python/rtdip_sdk/pipelines/sources/python/delta.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    
+    def post_read_validation(self):
+        return True
+    
+    def read_batch(self) -> LazyFrame:
+        '''
+        Reads data from a Delta table into a Polars LazyFrame
+        '''
+        without_files_dict = {"without_files": self.without_files}
+        lf = pl.scan_delta(source= self.path, version= self.version, storage_options= self.storage_options, delta_table_options= without_files_dict, pyarrow_options= self.pyarrow_options)
```
Comment (GBBBAS) [inline src/sdk/python/rtdip_sdk/pipelines/sources/python/delta.py:73]: is there a reason we don't offer delta_table_options in this component?

File: src/sdk/python/rtdip_sdk/pipelines/destinations/python/delta.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    def post_write_validation(self):
+        return True
+
+    def write_batch(self):
+        '''
+        Writes batch data to Delta without using Spark.
+        '''  
+        if isinstance(self.data, pl.LazyFrame):
+            df = self.data.collect() 
+            df.write_delta(self.path, mode=self.mode, overwrite_schema= self.overwrite_schema, storage_options=self.options, delta_write_options={"overwrite_schema": self.overwrite_schema})
```
Comment (GBBBAS) [inline src/sdk/python/rtdip_sdk/pipelines/destinations/python/delta.py:81]: should delta_write_options be provided as parameters to the component?

File: src/sdk/python/rtdip_sdk/pipelines/sources/python/delta.py (hunk 2 above)
Comment (JamesKnBr) [reply src/sdk/python/rtdip_sdk/pipelines/sources/python/delta.py:73]: From what I could see without_files was the only option you can have but I can change it

Comment (JamesKnBr) [reply src/sdk/python/rtdip_sdk/pipelines/sources/python/delta.py:73]: This is from the polars code

Comment (JamesKnBr) [reply src/sdk/python/rtdip_sdk/pipelines/sources/python/delta.py:73]: https://delta-io.github.io/delta-rs/python/api_reference.html#deltalake.table.DeltaTable

**Review by GBBBAS — APPROVED** [review]

**Pull request merged**